core/src/display_object/movie_clip.rs

```diff
@@ -99,6 +100,14 @@ pub struct MovieClipData<'gc> {
     queued_script_frame: Option<FrameNumber>,
     queued_goto_frame: Option<FrameNumber>,
     drop_target: Option<DisplayObject<'gc>>,
+
+    /// List of tags queued up for the current frame.
+    ///
+    /// There are only a handful of valid tag configurations per depth: namely,
+    /// the empty set, one removal, one add, or a removal followed by an add.
+    /// Any other configuration in the SWF tag stream is normalized to one of
+    /// these patterns.
+    queued_tags: HashMap<Depth, Vec<QueuedTag>>,
```

Comment:
would an enum `Empty, Add(...), Remove(...), AddRemove(...)` or `struct { add: Option, remove: Option }` instead be a bit simpler here? Might be able to remove vector iteration and some `unreachable`s then

Reply:
The struct option isn't any simpler than a Vec, aside from just not having a separate allocation (I probably should have used a SmallVec here). We still need to prohibit pairs of `Add` and `Remove` *in that order*, except the struct parameters are unordered so we have to sort by `tag_start` instead.

The enum option communicates the invariants in the actual type correctly, and the order can be implied in the enum type, so I'm probably going to go with that.

Reply:
The enum option worked great!